lms/envs/common.py

```diff
@@ -587,7 +587,7 @@
 # Note: these intentionally greedily grab all chars up to the next slash including any pluses
 # DHM: I really wanted to ensure the separators were the same (+ or /) but all patts I tried had
 # too many inadvertent side effects :-(
-COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/]+)'
+COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/][^?/]*)'
```

Comment:
@cpennington Without ```*``` it was not picking ```course_id``` if there is nothing at the end, like this ```http://foo.bar.com/courses/edx/Demo/2014```